### ISSUE TITLE
Invalid value for type="disk_io_time"

### ISSUE THREAD

**Issue by ghost** [logstash-plugins/logstash-codec-collectd#21]
Please post all product and debugging questions on our [forum](https://discuss.elastic.co/c/logstash). Your questions will reach our wider community members there, and if we confirm that there is a bug, then we can open a new issue here.

For all general issues, please provide the following details for fast resolution:
- Version: logstash-codec-collectd-2.0.4 running with logstash-2.3.4
- Operating System: Ubuntu 14.04.4 LTS (trusty)
- Config File (if you have sensitive info, please remove it):
- Sample Data: collectd-plugin-disk new metrics disk_io_time 
- Step to Reproduce, in the client side is running : 
  collectd          version 5.5.1.161
  collectd-core  version 5.5.1.161

**Comment by ghost** [comment]
`{:timestamp=>"2016-07-11T16:28:45.528000+0000", :message=>"Invalid value for type=\"disk_io_time\", key=nil, index=0", :level=>:error}
{:timestamp=>"2016-07-11T16:28:45.528000+0000", :message=>"Invalid value for type=\"disk_io_time\", key=nil, index=1", :level=>:error}
{:timestamp=>"2016-07-11T16:28:45.528000+0000", :message=>"Invalid value for type=\"disk_io_time\", key=nil, index=1", :level=>:error}
{:timestamp=>"2016-07-11T16:28:49.405000+0000", :message=>"Invalid value for type=\"disk_io_time\", key=nil, index=0", :level=>:error}
{:timestamp=>"2016-07-11T16:28:49.406000+0000", :message=>"Invalid value for type=\"disk_io_time\", key=nil, index=1", :level=>:error}
{:timestamp=>"2016-07-11T16:28:49.407000+0000", :message=>"Invalid value for type=\"disk_io_time\", key=nil, index=0", :level=>:error}
{:timestamp=>"2016-07-11T16:28:49.408000+0000", :message=>"Invalid value for type=\"disk_io_time\", key=nil, index=1", :level=>:error}
{:timestamp=>"2016-07-11T16:28:49.411000+0000", :message=>"Invalid value for type=\"disk_io_time\", key=nil, index=0", :level=>:error}
{:timestamp=>"2016-07-11T16:28:49.411000+0000", :message=>"Invalid value for type=\"disk_io_time\", key=nil, index=0", :level=>:error}
{:timestamp=>"2016-07-11T16:28:49.412000+0000", :message=>"Invalid value for type=\"disk_io_time\", key=nil, index=1", :level=>:error}
{:timestamp=>"2016-07-11T16:28:49.412000+0000", :message=>"Invalid value for type=\"disk_io_time\", key=nil, index=1", :level=>:error}
{:timestamp=>"2016-07-11T16:28:49.416000+0000", :message=>"Invalid value for type=\"disk_io_time\", key=nil, index=0", :level=>:error}`

**Comment by untergeek** [comment]
Can you send a sample of what collectd is sending?  Perhaps text-based output somewhere else?  You might need to set up something to describe the values better in a different typesdb file.

**Comment by ghost** [comment]
You are totally right, I included the new types.db coming with the new version of collectd 5.5.1 and it works fine . 
I discovered that the types.db version included in the plugin logstash-codec-collectd-2.0.4 is for collectd 5.4.0 . 
`# If no types.db is provided the included`types.db`will be used (currently 5.4.0).`

I made a diff between the 2 files and there are just a few new values. 
Maybe it would be better to include an higher version of this file (types.db).

**Comment by untergeek** [comment]
Might be better to just submit a PR with the newest version of typesdb, and update the doc to reflect that.

**Comment by anujsrc** [comment]
You can now specify the ``typesdb`` location as mentioned here- https://www.elastic.co/guide/en/logstash/current/plugins-codecs-collectd.html

**Comment by referup-tarantegui** [comment]
I've got the same issue. What is the proposed solution or workaround? I'm using the same version of logstash-codec-collectd-2.0.4 and logstash 2.4.
I'm using the default types.db included.

**Comment by pataquets** [comment]
any progress?
